squidb-processor/src/com/yahoo/squidb/processor/plugins/defaults/properties/TableModelSpecFieldPlugin.java

```diff
@@ -125,22 +159,139 @@ private boolean isIntegerPrimaryKey(VariableElement field, DeclaredTypeName fiel
+                .setModifiers(Modifier.PUBLIC)
+                .setReturnType(TypeConstants.LONG_PROPERTY)
+                .setMethodName("getRowIdProperty");
+        writer.beginMethodDefinition(params);
+        writer.writeStringStatement("return " + propertyName);
+        writer.finishMethodDefinition();
+    }
+
+    @Override
+    public void beforeEmitGetter(JavaFileWriter writer, PropertyGenerator propertyGenerator,
+            MethodDeclarationParameters getterParams) throws IOException {
+        addAccessorDocumentationForRowids(writer, propertyGenerator, true);
+    }
+
+    @Override
+    public void beforeEmitSetter(JavaFileWriter writer, PropertyGenerator propertyGenerator,
+            MethodDeclarationParameters setterParams) throws IOException {
+        addAccessorDocumentationForRowids(writer, propertyGenerator, false);
+    }
+
+    private void addAccessorDocumentationForRowids(JavaFileWriter writer, PropertyGenerator propertyGenerator,
+            boolean getter) throws IOException {
+        if (propertyGenerator instanceof RowidPropertyGenerator) {
+            if (DEFAULT_ROWID_PROPERTY_NAME.equals(propertyGenerator.getPropertyName())) {
+                writer.writeAnnotation(CoreTypes.OVERRIDE);
+            } else {
+                writer.writeJavadoc(" This " + (getter ? "getter" : "setter") + " is an alias for " +
+                        (getter ? "get" : "set") + "RowId(), as the underlying column is an INTEGER PRIMARY KEY");
+            }
+        }
+    }
+
+    @Override
+    public void emitMethods(JavaFileWriter writer) throws IOException {
+        // If rowid property generator hasn't already done it, need to generate
+        // overridden setRowId with appropriate return type
+        if (!pluginEnv.hasSquidbOption(PluginEnvironment.OPTIONS_DISABLE_DEFAULT_GETTERS_AND_SETTERS)) {
+            RowidPropertyGenerator rowidPropertyGenerator = modelSpec
+                    .getMetadata(METADATA_KEY_ROWID_ALIAS_PROPERTY_GENERATOR);
+            if (rowidPropertyGenerator != null && !"setRowId".equals(rowidPropertyGenerator.setterMethodName())) {
+                MethodDeclarationParameters params = new MethodDeclarationParameters()
+                        .setModifiers(Modifier.PUBLIC)
+                        .setMethodName("setRowId")
+                        .setArgumentTypes(CoreTypes.PRIMITIVE_LONG)
+                        .setArgumentNames("rowid")
+                        .setReturnType(modelSpec.getGeneratedClassName());
+                writer.writeAnnotation(CoreTypes.OVERRIDE)
+                        .beginMethodDefinition(params)
+                        .writeStringStatement("super.setRowId(rowid)")
+                        .writeStringStatement("return this")
+                        .finishMethodDefinition();
+            }
         }
     }
 
+    // Users can subclass TableModelSpecFieldPlugin and override these methods to handle the core types differently
```

Comment:
We should probably write some javadoc for these methods in case someone wants to do this